### PR TITLE
Update "What is KPOps" section to be more catchy

### LINKED ISSUE #143: Update "What is KPOps" section to be more catchy

### PR DESCRIPTION
**Pull request by sujuka99**
Closes #143

### CODE REVIEW

File: docs/docs/resources/pipeline-components/kafka-source-connector.yaml [hunk 1]
```diff
@@ -40,4 +39,5 @@
   # Image Tag etc.
   resetterValues:
     imageTag: 1.2.3
+  #  https://kafka.apache.org/documentation/#impl_offsettracking
```
Comment (philipp94831) [inline docs/docs/resources/pipeline-components/kafka-source-connector.yaml:42]: That is actually the wrong link. This refers to the standard Kafka Consumer group offsets. However, here we refer to Kafka Connect offsets for source connectors, which is quite a bit different

**Review by philipp94831 — CHANGES_REQUESTED** [review]

File: docs/docs/resources/pipeline-components/kafka-source-connector.yaml (hunk 1 above)
Comment (sujuka99) [reply docs/docs/resources/pipeline-components/kafka-source-connector.yaml:42]: Is it [offset.storage.topic](https://kafka.apache.org/documentation/#connect_running) or [offset.storage.file.filename](https://kafka.apache.org/documentation/#connect_running)?

File: docs/docs/user/what-is-kpops.md [hunk 2]
```diff
@@ -1,7 +1,22 @@
 # What is KPOps?
 
-KPOps is a CLI tool that allows deploying and destroying consecutive Kafka Streams applications, Producers, and Kafka 
-Connectors using an easy-to-read and write pipeline definition. KPOps comes with various abstractions that can simplify
-configuring multiple pipelines and steps within pipelines by sharing configuration between different applications,
-like producers or streaming applications. In addition, Kafka apps and its resources can be reset or cleaned up 
-again with a single command.
+With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can [`destroy`](/user/references/cli-commands/#deploy), [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`deploy`](/user/references/cli-commands/#deploy) a pipeline[^1] to a Kubernetes cluster!
```
Comment (disrupted) [inline docs/docs/user/what-is-kpops.md:3]:
```suggestion
With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can [`deploy`](/user/references/cli-commands/#deploy), [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) a Kafka pipeline[^1] to a Kubernetes cluster!
```

**Review by disrupted — CHANGES_REQUESTED** [review]
I like it 🙂

File: docs/docs/user/what-is-kpops.md [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+
+- Deploy pipelines with minimal effort and without sacrificing configurability
+- Easy-to-read and write pipeline definitions
+- Reset or destroy pipelines gracefully
+- Reset or clean apps again with a single command
+- Set defaults to share settings between pipeline steps (or whole different pipelines!)
+
+## Example
+
+<!-- Replace with https://raw.githubusercontent.com/bakdata/kpops-examples/main/word-count/deployment/kpops/pipeline.yaml -->
```
[the remark this answers is not in the document]
Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md]: done

File: docs/docs/user/what-is-kpops.md (hunk 2 above)
Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md]: I had it like that initially, but I think it doesn't sound good: "...destroy a Kafka pipeline to a Kubernetes cluster!". It seems I forgot to switch out the links tho..

What do you think of it now? I reworded it a bit

File: docs/docs/user/what-is-kpops.md [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+[^1]:
+     A pipeline can consist of just consecutive [Kafka Streams applications](/user/references/components/#streamsapp) or it could also contain [Producers](/user/references/components/#producer) and/or [Kafka Connectors](/user/references/defaults/#kafkaconnector).
+
+## Key features
+
+- Deploy pipelines with minimal effort and without sacrificing configurability
+- Easy-to-read and write pipeline definitions
+- Reset or destroy pipelines gracefully
+- Reset or clean apps again with a single command
+- Set defaults to share settings between pipeline steps (or whole different pipelines!)
```
Comment (raminqaf) [inline docs/docs/user/what-is-kpops.md:13]: Maybe link the default documentation here.

Comment (raminqaf) [inline docs/docs/user/what-is-kpops.md:13]: Should we update the `Future` section of the readme with these points?

Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md:13]: Im not sure tbh, do you think they are better? If not, I could instead replace these with the points from the README.

File: docs/docs/user/what-is-kpops.md [hunk 5]
```diff
@@ -1,7 +1,26 @@
 # What is KPOps?
 
-KPOps is a CLI tool that allows deploying and destroying consecutive Kafka Streams applications, Producers, and Kafka 
-Connectors using an easy-to-read and write pipeline definition. KPOps comes with various abstractions that can simplify
-configuring multiple pipelines and steps within pipelines by sharing configuration between different applications,
-like producers or streaming applications. In addition, Kafka apps and its resources can be reset or cleaned up 
-again with a single command.
+With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
```
Comment (disrupted) [inline docs/docs/user/what-is-kpops.md:3]:
```suggestion
With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
```

File: docs/docs/user/what-is-kpops.md [hunk 6]
```diff
@@ -1,7 +1,26 @@
...
+[^1]:
+     A pipeline can consist of just consecutive [Kafka Streams applications](/user/references/components/#streamsapp) or it could also contain [Producers](/user/references/components/#producer) and/or [Kafka Connectors](/user/references/defaults/#kafkaconnector).
```
Comment (disrupted) [inline docs/docs/user/what-is-kpops.md:5]:
```suggestion
     A Kafka pipeline can consist of just consecutive [streaming applications](/user/references/components/#streamsapp), [producers](/user/references/components/#producer) and [connectors](/user/references/defaults/#kafkaconnector).
```

Comment (disrupted) [reply docs/docs/user/what-is-kpops.md:5]: I would prefer this wording, what do you think?

**Review by disrupted — DISMISSED** [review]

File: docs/docs/user/what-is-kpops.md (hunk 6 above)
Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md:5]: I agree.

File: docs/docs/user/what-is-kpops.md [hunk 7]
```diff
@@ -1,7 +1,26 @@
...
+- Easy-to-read and write pipeline definitions
+- Reset or destroy pipelines gracefully
+- Reset or clean apps again with a single command
+- Set defaults to share settings between pipeline steps (or whole different pipelines!)
+
+## Example
+
+<figure markdown>
+  ![atm-fraud-pipeline](/images/word-count-pipeline_streams-explorer.png)
+  <figcaption>An overview of <a href="/user/getting-started/quick-start">Word-count pipeline</a> shown in <a href="https://github.com/bakdata/streams-explorer#streams-explorer" target="_blank">Streams Explorer</a></figcaption>
```
[the remark this answers is not in the document]
Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md:19]: Are you sure? 

Word-count is the name of the pipeline, to me it makes sense to capitalize it. It is also capitalized everywhere else in the documentation.

File: docs/docs/user/what-is-kpops.md (hunk 4 above)
Comment (raminqaf) [reply docs/docs/user/what-is-kpops.md]: Actually not! I think both of the sections provide good insights about KPOps keept it like that

File: docs/docs/user/what-is-kpops.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (raminqaf) [reply docs/docs/user/what-is-kpops.md:19]: Okay! If you say that it is a name and you kept it like that everywhere in the docs I get that.  So fine for my side.

File: docs/docs/user/what-is-kpops.md [hunk 8]
```diff
@@ -1,7 +1,26 @@
 # What is KPOps?
 
-KPOps is a CLI tool that allows deploying and destroying consecutive Kafka Streams applications, Producers, and Kafka 
-Connectors using an easy-to-read and write pipeline definition. KPOps comes with various abstractions that can simplify
-configuring multiple pipelines and steps within pipelines by sharing configuration between different applications,
-like producers or streaming applications. In addition, Kafka apps and its resources can be reset or cleaned up 
-again with a single command.
+With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
+[^1]:
```
Comment (raminqaf) [inline docs/docs/user/what-is-kpops.md:4]:
```suggestion
With a couple of easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
[^1]:
```

File: docs/docs/user/what-is-kpops.md [hunk 9]
```diff
@@ -1,7 +1,26 @@
...
+     A Kafka pipeline can consist of consecutive [streaming applications](/user/references/components/#streamsapp), [producers](/user/references/components/#producer), and [connectors](/user/references/defaults/#kafkaconnector).
+
+## Key features
+
+- Deploy pipelines with minimal effort and without sacrificing configurability
+- Easy-to-read and write pipeline definitions
+- Reset or destroy pipelines gracefully
+- Reset or clean apps again with a single command
+- Set [defaults](/user/references/defaults) to share settings between pipeline steps (or whole different pipelines!)
```
Comment (chrboehm) [inline docs/docs/user/what-is-kpops.md:13]: share settings within and across pipelines

File: docs/docs/user/what-is-kpops.md [hunk 10]
```diff
@@ -1,7 +1,26 @@
...
-like producers or streaming applications. In addition, Kafka apps and its resources can be reset or cleaned up 
-again with a single command.
+With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
+[^1]:
+     A Kafka pipeline can consist of consecutive [streaming applications](/user/references/components/#streamsapp), [producers](/user/references/components/#producer), and [connectors](/user/references/defaults/#kafkaconnector).
+
+## Key features
+
+- Deploy pipelines with minimal effort and without sacrificing configurability
+- Easy-to-read and write pipeline definitions
```
Comment (chrboehm) [inline docs/docs/user/what-is-kpops.md:10]: would use this as point 1: Easy-to-read and -write ...

Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md:9]: fixed

File: docs/docs/user/what-is-kpops.md (hunk 9 above)
Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md]: fixed

File: docs/docs/user/what-is-kpops.md (hunk 10 above)
Comment (disrupted) [reply docs/docs/user/what-is-kpops.md:9]: don't forget to add the leading hyphen in _-write_ as proposed by Christoph

Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md]: good point, done

File: docs/docs/user/what-is-kpops.md [hunk 11]
```diff
@@ -1,7 +1,26 @@
...
+[^1]:
+     A Kafka pipeline can consist of consecutive [streaming applications](/user/references/components/#streamsapp), [producers](/user/references/components/#producer), and [connectors](/user/references/defaults/#kafkaconnector).
+
+## Key features
+
+- Easy-to-read and -write pipeline definitions
+- Deploy pipelines with minimal effort and without sacrificing configurability
+- Reset or destroy pipelines gracefully
+- Reset or clean apps again with a single command
+- Set [defaults](/user/references/defaults) to share settings within and across pipelines
```
Comment (raminqaf) [inline docs/docs/user/what-is-kpops.md:13]: @disrupted WDYT?
```suggestion
- **Deploy Kafka apps to Kubernetes**: KPOps allows to deploy consecutive Kafka Streams applications and producers using an easy-to-read and write pipeline definition.
- **Manage Kafka Connectors**: KPOps connects with your Kafka connect cluster and applies, validates, and deletes your connector configs.
- **Configure multiple pipelines and steps**: KPOps has various abstractions that simplify configuring multiple pipelines and steps within pipelines by sharing configuration between different applications, like producers or streaming applications.
- **Handle your topics and schemas**: KPOps not only creates and deletes your topics but also registers and deletes your schemas.
- **Manage the lifecycle of your components**: KPOps not only terminates your pipeline components (i.e., Kafka Streams applications) from your Kubernetes cluster, but it cleans up the (Kafka) component-related states (i.e., removing/resetting offset of consumer groups).
```

Comment (disrupted) [reply docs/docs/user/what-is-kpops.md:13]: @raminqaf looks great!

Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md:13]: I don't like the repeating "not only" and I am not sure i **manage the lifecycle of your components** is an accurate title for that point as it only talks about destruction.

```suggestion
- **Deploy Kafka apps to Kubernetes**: KPOps allows to deploy consecutive Kafka Streams applications and producers using an easy-to-read and write pipeline definition.
- **Manage Kafka Connectors**: KPOps connects with your Kafka connect cluster and applies, validates, and deletes your connector configs.
- **Configure multiple pipelines and steps**: KPOps has various abstractions that simplify configuring multiple pipelines and steps within pipelines by sharing configuration between different applications, like producers or streaming applications.
- **Handle your topics and schemas**: KPOps not only creates and deletes your topics but also registers and deletes your schemas.
- **Remove your components**: KPOps removes your pipeline components (i.e., Kafka Streams applications) from the Kubernetes cluster AND cleans up the (Kafka) component-related states (i.e., removing/resetting offset of consumer groups).
- **Preview your pipeline changes**: With the KPOps dry-run, you can ensure your pipeline definition is set up correctly. This helps to minimize downtime and prevent potential errors or issues that could impact your production environment.
```

Comment (raminqaf) [reply docs/docs/user/what-is-kpops.md:13]: I like it now even more. How about we change **Remove your components** to **Clean termination of Kafka components**

Comment (sujuka99) [reply docs/docs/user/what-is-kpops.md:13]: Even better! :) 

@disrupted wdyt?

Comment (disrupted) [reply docs/docs/user/what-is-kpops.md]: great improvements all around, go for it!

File: docs/docs/user/what-is-kpops.md [hunk 12]
```diff
@@ -1,7 +1,27 @@
...
-configuring multiple pipelines and steps within pipelines by sharing configuration between different applications,
-like producers or streaming applications. In addition, Kafka apps and its resources can be reset or cleaned up 
-again with a single command.
+With a couple of easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
+[^1]:
+     A Kafka pipeline can consist of consecutive [streaming applications](/user/references/components/#streamsapp), [producers](/user/references/components/#producer), and [connectors](/user/references/defaults/#kafkaconnector).
+
+## Key features
+
+- **Deploy Kafka apps to Kubernetes**: KPOps allows to deploy consecutive Kafka Streams applications and producers using an easy-to-read and write pipeline definition.
```
Comment (disrupted) [inline docs/docs/user/what-is-kpops.md:9]:
```suggestion
- **Deploy Kafka apps to Kubernetes**: KPOps allows to deploy consecutive Kafka Streams applications and producers using an easy-to-read and -write pipeline definition.
```
this somehow got changed again

File: docs/docs/user/what-is-kpops.md [hunk 13]
```diff
@@ -1,7 +1,27 @@
...
+- **Manage Kafka Connectors**: KPOps connects with your Kafka connect cluster and applies, validates, and deletes your connector configs.
+- **Configure multiple pipelines and steps**: KPOps has various abstractions that simplify configuring multiple pipelines and steps within pipelines by sharing configuration between different applications, like producers or streaming applications.
```
Comment (disrupted) [inline docs/docs/user/what-is-kpops.md:11]:
```suggestion
- **Configure multiple pipelines and steps**: KPOps has various abstractions that simplify configuring multiple pipelines and steps within pipelines by sharing common configuration between different components, such as producers or streaming applications.
```

File: docs/docs/user/what-is-kpops.md [hunk 14]
```diff
@@ -1,7 +1,27 @@
...
+- **Handle your topics and schemas**: KPOps not only creates and deletes your topics but also registers and deletes your schemas.
+- **Clean termination of Kafka components**: KPOps removes your pipeline components (i.e., Kafka Streams applications) from the Kubernetes cluster AND cleans up the (Kafka) component-related states (i.e., removing/resetting offset of consumer groups).
```
Comment (disrupted) [inline docs/docs/user/what-is-kpops.md:13]:
```suggestion
- **Clean termination of Kafka components**: KPOps removes your pipeline components (i.e., Kafka Streams applications) from the Kubernetes cluster _and_ cleans up the component-related states (i.e., removing/resetting offset of Kafka consumer groups).
```

**Review by raminqaf — APPROVED** [review]

**Pull request merged**